pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -162,32 +164,75 @@ func (r *ReconcileKogitoApp) Reconcile(request reconcile.Request) (reconcile.Res
 		},
 	})
 	if err != nil {
-		return reconcile.Result{}, err
+		updateResourceResult.Err = err
+		updateResourceResult.ErrorReason = v1alpha1.ParseCRRequestFailedReason
 	}
 
-	if kogitoResources.BuildConfigS2IStatus.IsNew {
-		log.Infof("Buildconfigs are created, triggering build %s", kogitoResources.BuildConfigS2I.Name)
-		if _, err := openshift.BuildConfigC(r.client).TriggerBuild(kogitoResources.BuildConfigS2I, instance.Name); err != nil {
-			return reconcile.Result{}, err
+	deployedRes, err := kogitores.GetDeployedResources(instance, r.client.ControlCli)
+	if err != nil {
+		updateResourceResult.Err = err
+		updateResourceResult.ErrorReason = v1alpha1.RetrieveDeployedResourceFailedReason
+	}
+
+	requestedRes := compare.NewMapBuilder().Add(k8sRes...).ResourceMap()
+
+	comparator := kogitores.GetComparator()
+	deltas := comparator.Compare(deployedRes, requestedRes)
+
+	writer := write.New(r.client.ControlCli).WithOwnerController(instance, r.scheme)
+
+	var hasUpdates bool
```

Comment:
This is the logic I meant by having all the status/update handling in one place.